test/nrepl_refactor.vim

```diff
@@ -319,4 +319,64 @@ function! s:suite.add_arity_defmethod_test() abort
 endfunction
 " }}}
 
+" rename_symbol {{{
+function! s:suite.rename_symbol_test() abort
+  let def_file = tempname()
+  call writefile([
+        \'(ns user)',
+        \'   (defn bar [] :bar)'], def_file)
+
+  let alias_file = tempname()
+  call writefile([
+        \'(ns a (:require [user :as a.u])',
+        \';; bar',
+        \'(let [bar (a.u/bar :bar))]'], alias_file)
+
+  let refer_file = tempname()
+  call writefile([
+        \'(ns a (:require [user :refer [bar]])',
+        \';; bar',
+        \'(let [bar (bar :bar))]'], refer_file)
+
+  let responses = {}
+  let responses['info'] = {
+        \'ns': 'user',
+        \'name': 'bar',
+        \'file': def_file,
+        \'line': 2,
+        \'column': 1,
+        \'status': ['done']}
+  let responses['find-symbol'] = [
+        \{'occurrence': '{:file "'.def_file.'"   :line-beg 2 :col-beg 4}'},
+        \{'occurrence': '{:file "'.alias_file.'" :line-beg 3 :col-beg 13}'},
+        \{'occurrence': '{:file "'.refer_file.'" :line-beg 1 :col-beg 31}'},
+        \{'occurrence': '{:file "'.refer_file.'" :line-beg 3 :col-beg 12}'},
+        \{'status': ['done']}]
+  call s:ch.mock({'status_value': 'open', 'relay':
+        \{m -> get(responses, m['op'], {'status': ['done']})}})
+
+
+  call s:io.mock({'input': 'new-name'})
+  call iced#nrepl#refactor#rename_symbol('dummy')
+  sleep 1000m " TODO figure out how to wait for finished refactoring
```

Comment:
@liquidz could you please help me to figure out how to run the test without sleep?
It looks like without the sleep assertions runs before refactoring have chance to write a file.

Reply:
When `iced#nrepl#refactor#rename_symbol` returns a Promise object, you can wait finishing refactoring as below.
```vim
let p = iced#nrepl#refactor#rename_symbol('dummy')
call iced#promise#wait(p)
```